src/Umbraco.Infrastructure/Persistence/Repositories/Implement/RelationRepository.cs

```diff
@@ -201,9 +233,29 @@ public IEnumerable<IUmbracoEntity> GetPagedChildEntitiesByParentId(int parentId,
 
                 sql.Where<RelationDto>(rel => rel.ParentId == parentId);
                 sql.Where<RelationDto, NodeDto>((rel, node) => rel.ChildId == parentId || node.NodeId != parentId);
+
+                if (relationTypes != null && relationTypes.Any())
+                {
+                    sql.WhereIn<RelationDto>(rel => rel.RelationType, relationTypes);
+                }
             });
         }
 
+        public IEnumerable<IUmbracoEntity> GetPagedEntitiesForItemsInRelation(int[] itemIds, long pageIndex, int pageSize, out long totalRecords, params Guid[] entityTypes)
+        {
+            return _entityRepository.GetPagedResultsByQuery(Query<IUmbracoEntity>(), entityTypes, pageIndex, pageSize, out totalRecords, null, null, sql =>
+            {
+                SqlJoinRelations(sql);
+
+                sql.WhereIn<RelationDto>(rel => rel.ChildId, itemIds);
+                sql.Where<RelationDto, NodeDto>((rel, node) => rel.ChildId == node.NodeId);
+                sql.Where<RelationTypeDto>(type => type.IsDependency);
```

Comment:
This query would benefit from an index on ChildId, IsDependency on RelationDto

Reply:
Good point.. Will create task for that

Reply:
`RelationRepository.GetPagedEntitiesForItemsInRelation` is not used as the logic has been moved to a different repo but we will create the indexes as they are still useful for the use case 👍

Reply:
The new place where we are making a similar query is [TrackedReferencesRepository.GetPagedItemsWithRelations()](https://github.com/umbraco/Umbraco-CMS/pull/12146/files#diff-f2cfcf8ca0f4c9e27e08abcf15a90a7b1081df4a9a73ff932d5e7aeeb2dc6875R25)